### PR TITLE
fix: cumulative payment dynamo db unit conversion

### PR DESCRIPTION
**Pull request by hopeyen**
## Why are these changes needed?

Quick PR to move away from using dynamoDB `cumulativePayment` value to `uint64` using `strconv.ParseUint`; directly convert to big.Int

## Checks

- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, in that case, please comment that they are not relevant.
- [ ] I've checked the new test coverage and the coverage percentage didn't drop.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Integration tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: core/meterer/meterer.go [hunk 1]
```diff
@@ -230,20 +231,20 @@ func (m *Meterer) ValidatePayment(ctx context.Context, header core.PaymentMetada
...
 	}
 	// check passed: blob can be safely inserted into the set of payments
 	return nil
 }
 
 // PaymentCharged returns the chargeable price for a given data length
-func (m *Meterer) PaymentCharged(numSymbols uint) uint64 {
-	return uint64(m.SymbolsCharged(numSymbols)) * uint64(m.ChainPaymentState.GetPricePerSymbol())
+func (m *Meterer) PaymentCharged(numSymbols uint) *big.Int {
+	return big.NewInt(int64(m.SymbolsCharged(numSymbols) * m.ChainPaymentState.GetPricePerSymbol()))
```
[the remark this answers is not in the document]
Comment (hopeyen) [reply core/meterer/meterer.go:247]: Highly unlikely 😬 If we induct backwards to find what's allowed by uint64 with our expected price per symbol `447,000,000` then the number of symbols uint64 can charge is `2^64/477,000,000=41,267,883,833.8`. With 1 symbol being 32 bytes, we are going to allow `1,320,572,282,681.7bytes=1.3TB` per blob

The reason I updated this is for consistency in the `cumulativePayment` calculations, but I'm happy to revert these changes and just convert the numbers to big.Int if you think that's better?

**Comment by hopeyen** [comment]
> does this PR have the changes made on dynamodb storage?

no, I think the changes are only how we read from dynamodb. Writing to the db is still the same as before

File: core/meterer/offchain_store.go [hunk 2]
```diff
@@ -199,14 +180,15 @@ func (s *OffchainStore) GetRelevantOnDemandRecords(ctx context.Context, accountI
...
-		return 0, 0, 0, fmt.Errorf("failed to query smaller payments for account: %w", err)
+		return nil, nil, 0, fmt.Errorf("failed to query smaller payments for account: %w", err)
 	}
-	var prevPayment uint64
+	prevPayment := big.NewInt(0)
 	if len(smallerResult) > 0 {
-		prevPayment, err = strconv.ParseUint(smallerResult[0]["CumulativePayments"].(*types.AttributeValueMemberN).Value, 10, 64)
-		if err != nil {
-			return 0, 0, 0, fmt.Errorf("failed to parse previous payment: %w", err)
+		setPrevPayment, success := prevPayment.SetString(smallerResult[0]["CumulativePayments"].(*types.AttributeValueMemberN).Value, 10)
```
Comment (ian-shim) [inline core/meterer/offchain_store.go:187]: I know these values are all expected in the happy path, but this chain seems dangerous
It's missing at least a couple validations: checking if `"CumulativePayments"` exists in the map and the result of the type assertion

Here and below as well

File: core/meterer/meterer.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (ian-shim) [reply core/meterer/meterer.go:247]: Yeah converting each values to big int and using `num0.Mul(num1)` seems more safe?

File: core/meterer/offchain_store.go (hunk 2 above)
Comment (hopeyen) [reply core/meterer/offchain_store.go]: added explicit checking for them!

**Review by ian-shim — APPROVED** [review]

**Pull request merged**